src/charts/loading/LoadingContainer.js

```diff
@@ -0,0 +1,73 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+export const loadingContainerWrapper = ({data, chart, shouldShowLoadingState}, container) => {
+    if (shouldShowLoadingState) {
+        container = (
+            <LoadingContainer
+                data={data}
+                loadingState={chart.loading()}
+            >
+                {container}
+            </LoadingContainer>
+        );
+    }
+    return container;
+}
+
+const toggleLoading = (state) => ({loading: !state.loading});
+
+export default class LoadingContainer extends PureComponent {
+    static propTypes = {
+        children: PropTypes.element,
+        data: PropTypes.oneOfType(
+            [
+                PropTypes.object,
+                PropTypes.array,
+            ]
+        ),
+        loadingState: PropTypes.string,
+    }
+
+    state = {
+        loading: true,
+    }
+
+    componentWillMount() {
+        if (this.props.data !== null) {
+            this.setState(toggleLoading);
+        }
+    }
```

Comment:
Have you checked to see if this will work with `core`?

[`componentWillMount` gets called during server side rendering](https://reactjs.org/docs/react-component.html#componentwillmount) so will this cause the server to attempt to render the chart? Which would probably fail? Not sure.

Reply:
Good call, will try it in core and comment with results.

Reply:
Just switched to `componentDidMount` since we know that works and since the loading state is optional.

Reply:
This still says `componentWillMount`, are you still going to change it later?